centreon/tests/php/Core/HostCategory/Application/UseCase/AddHostCategory/AddHostCategoryTest.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Core\HostCategory\Application\UseCase\AddHostCategory;
+
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Core\Application\Common\UseCase\CreatedResponse;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+use Core\Application\Common\UseCase\InvalidArgumentResponse;
+use Core\HostCategory\Application\Exception\HostCategoryException;
+use Core\HostCategory\Application\Repository\ReadHostCategoryRepositoryInterface;
+use Core\HostCategory\Application\Repository\WriteHostCategoryRepositoryInterface;
+use Core\HostCategory\Application\UseCase\AddHostCategory\AddHostCategory;
+use Core\HostCategory\Application\UseCase\AddHostCategory\AddHostCategoryRequest;
+use Core\HostCategory\Domain\Model\HostCategory;
+use Core\Infrastructure\Common\Api\DefaultPresenter;
+use Core\Infrastructure\Common\Presenter\PresenterFormatterInterface;
+
+beforeEach(function () {
+    $this->writeHostCategoryRepository = $this->createMock(WriteHostCategoryRepositoryInterface::class);
+    $this->readHostCategoryRepository = $this->createMock(ReadHostCategoryRepositoryInterface::class);
+    $this->presenterFormatter = $this->createMock(PresenterFormatterInterface::class);
+    $this->user = $this->createMock(ContactInterface::class);
+    $this->request = new AddHostCategoryRequest();
+    $this->request->name = 'hc-name';
+    $this->request->alias = 'hc-alias';
+    $this->request->comment = null;
+    $this->presenter = new DefaultPresenter($this->presenterFormatter);
+    $this->useCase = new AddHostCategory(
+        $this->writeHostCategoryRepository,
+        $this->readHostCategoryRepository,
+        $this->user
+    );
+    $this->hostCategory = new HostCategory(1, $this->request->name, $this->request->alias);
+});
+
+it('should present an ForbiddenResponse when a user has unsufficient rights', function (): void {
```

Comment:
Missing tests for me
```
should present an ErrorResponse when a generic exception is thrown
should present an ErrorResponse if the just created host group cannot be retrieved
should present a CreatedResponse<AddHostCategoryResponse> as admin
should present a CreatedResponse<AddHostCategoryResponse> as allowed READ_WRITE user
should present a ForbiddenResponse as allowed READ user
```

Reply:
should present a CreatedResponse<AddHostCategoryResponse> as admin
should present a CreatedResponse<AddHostCategoryResponse> as allowed READ_WRITE user
should present a ForbiddenResponse as allowed READ user
should present an ErrorResponse when a generic exception is thrown
-> those cases are already handled with:
should present an ForbiddenResponse when a user has unsufficient rights
should return CreatedResponse on success
should present an ErrorResponse when an exception is thrown

should present an ErrorResponse if the just created host group cannot be retrieved
-> has been created